src/dataStructures.hpp

```diff
@@ -205,33 +205,158 @@ struct SpongeZoneData {
   bool singleTemperature;
 };
 
-struct volumeFaceIntegrationArrays {
-  // nodes IDs and indirection array
-  Array<int> nodesIDs;
-  Array<int> posDofIds;
-  // count of number of elements of each type
+/** @brief Array providing information about dof indices for elements
+ *
+ * These arrays store information about the degrees of freedom
+ * associated with each element.  They are used in the _GPU_ code path
+ * to provide information that would otherwise (i.e., on the cpu path)
+ * come from, for instance, a FiniteElement or FiniteElementSpace
+ * object.
+ */
+struct elementIndexingData {
+  /** Dof array offset for element i
+   *
+   *  To be used with element_dofs_list.  Specifically,
+   *
+   *  element_dofs_list[element_dof_offset[i] + j] = index of jth dof of element i
+   */
+  Array<int> element_dof_offset;
```

Comment:
Overall naming thought: probably can simplify names in these structs as the `element_` and `face_` prefixes are generally redundant with the name of the struct itself.

Reply:
3b8dcd4